util/argo/argo.go

```diff
@@ -455,3 +460,56 @@ func getDestinationServer(ctx context.Context, db db.ArgoDB, clusterName string)
 	}
 	return servers[0], nil
 }
+
+func GetAppVirtualProject(proj *argoappv1.AppProject, projLister applicationsv1.AppProjectLister, settingsManager *settings.SettingsManager) (*argoappv1.AppProject, error) {
+	gps, err := settingsManager.GetGlobalProjectsSettings()
+	if err != nil {
+		log.Warnf("Failed to get global project settings")
+		return proj, err
```

Comment:
We log a warning message but return an error here.   If a function returns an error, it should return nil for other non-error return values.   [See here](https://www.digitalocean.com/community/tutorials/handling-errors-in-go#:~:text=When%20a%20function%20returns%20multiple,value%20to%20a%20zero%20value) (and look for "returns multiple value")
It looks like the intention is that missing or malformed argocd-cm would fallback to the old `GetAppProject` behavior.  If this is the case, I would log a warning with including the message from `err` and `return proj, nil`.

Reply:
Great input! Thank you!